agent/engine/docker_image_manager.go

```diff
@@ -126,15 +128,34 @@ func (imageManager *dockerImageManager) RecordContainerReference(container *apic
 		seelog.Errorf("Error inspecting image %v: %v", container.Image, err)
 		return err
 	}
-
 	container.ImageID = imageInspected.ID
+	container.ImageDigest = imageManager.fetchRepoDigest(imageInspected, container)
 	added := imageManager.addContainerReferenceToExistingImageState(container)
 	if !added {
 		imageManager.addContainerReferenceToNewImageState(container, imageInspected.Size)
 	}
 	return nil
 }
 
+// The helper function to fetch the RepoImageDigest when inspect the image
+func (imageManager *dockerImageManager) fetchRepoDigest(imageInspected *types.ImageInspect, container *apicontainer.Container) string {
+	imageRepoDigests := imageInspected.RepoDigests
+	resultRepoDigest := ""
+	imagePrefix := strings.Split(container.Image, ":")[0]
+	for _, imageRepoDigest := range imageRepoDigests {
+		if strings.HasPrefix(imageRepoDigest, imagePrefix) {
+			repoDigestSplitList := strings.Split(imageRepoDigest, "@")
+			if len(repoDigestSplitList) > 1 {
+				resultRepoDigest = repoDigestSplitList[1]
+			} else {
+				seelog.Debugf("imageRepoDigest doesn't have the right format: [%s]", imageRepoDigest)
```

Comment:
You might want to log this as a warn when it doesn't have the format you are expecting.